src/modules/accounting/accounting.py

```diff
@@ -347,13 +371,108 @@ def _calculate_rebase_report(self, blockstamp: ReferenceBlockStamp) -> RebaseRep
         shares_requested_to_burn = self.get_shares_to_burn(blockstamp)
-        return validators_count, cl_balance, withdrawal_vault_balance, el_rewards_vault_balance, shares_requested_to_burn
+        return (
+            validators_count,
+            cl_balance,
+            withdrawal_vault_balance,
+            el_rewards_vault_balance,
+            shares_requested_to_burn,
+        )
 
-    # calculates is_bunker, finalization_share_rate, finalization_batches
+    # calculates is_bunker, finalization_batches
     def _calculate_wq_report(self, blockstamp: ReferenceBlockStamp) -> WqReport:
         is_bunker = self._is_bunker(blockstamp)
-        finalization_share_rate, finalization_batches = self._get_finalization_data(blockstamp)
-        return is_bunker, finalization_share_rate, finalization_batches
+        finalization_batches = self._get_finalization_batches(blockstamp)
+        return is_bunker, finalization_batches
+
+    def _handle_vaults_report(self, blockstamp: ReferenceBlockStamp) -> VaultsReport:
+        """
+        Generates and publishes a Merkle tree report for staking vaults at a given blockstamp.
+
+        This function performs the following steps:
+        1. Fetches staking vaults at the given block number.
+        2. If no vaults are found, returns empty report data.
+        3. Retrieves validator data and pending deposits at the refBlock.
+        4. Loads chain configuration for the target block.
+        5. Computes total values for all vaults, including validator balances and pending deposits.
+        6. Calculates vault-specific fees and slashing reserves.
+        7. Constructs structured tree data combining vault values, fees, and slashing reserves.
+        8. Builds a Merkle tree from this data.
+        9. Publishes the Merkle tree to IPFS (or similar) and retrieves a content ID (CID).
+        10. Returns the Merkle tree root and the CID as the vaults report.
+
+        Args:
+            blockstamp (ReferenceBlockStamp): Block context used to fetch vault and validator data.
+
+        Returns:
+            VaultsReport: A tuple containing the Merkle tree root (as `bytes`) and the CID (as `str`)
+                          identifying the published tree data.
+        """
+
+        vaults = self.staking_vaults.get_vaults(blockstamp.block_hash)
+        if len(vaults) == 0:
+            return bytes(0), ''
+
+        validators = self.w3.cc.get_validators(blockstamp)
+        pending_deposits = self.w3.cc.get_pending_deposits(blockstamp)
+        chain_config = self.get_chain_config(blockstamp)
+        frame_config = self.get_frame_config(blockstamp)
+        simulation = self.simulate_full_rebase(blockstamp)
+
+        core_apr_ratio = calculate_gross_core_apr(
+            pre_total_shares=simulation.pre_total_shares,
+            pre_total_ether=simulation.pre_total_pooled_ether,
+            post_total_shares=simulation.post_total_shares - simulation.shares_to_mint_as_fees,
```

Comment:
we need to use `postInternalShares` and `postInternalEther` here to calculate gross apr correctly